File: examples/Python/Frame/frame.py
from unicodedata import decimal
import os
from scisdk.scisdk import SciSDK
from scisdk.scisdk_defines import *
from struct import *


# initialize scisdk library
sdk = SciSDK()

#DT1260
res = sdk.AddNewDevice("usb:10500","dt1260", "./DT1260RegisterFile.json","board0")
#DT5560
#res = sdk.AddNewDevice("192.168.50.10:8888","DT5560", "./DT5560RegisterFile.json","board0")
#DT5550
#res = sdk.AddNewDevice("usb:11000","DT5550", "./DT5550RegisterFile.json","board0")
#V2740
#res = sdk.AddNewDevice("192.168.50.10","V2740", "./V2740RegisterFile.json","board0")

if res != 0:
    print ("Script exit due to connetion error")
    exit()

# set oscilloscope parameters

res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.thread", "false")
res = sdk.SetParameterInteger("board0:/MMCComponents/ImageReadout_0.timeout", 500)
res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.acq_mode", "non-blocking")
res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.trigger_mode", "trig")
res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.data_processing", "decode")

# allocate buffer raw, size 1024
res, buf = sdk.AllocateBuffer("board0:/MMCComponents/ImageReadout_0", 100)

res = sdk.ExecuteCommand("board0:/MMCComponents/ImageReadout_0.start", "")

while True:
    res, buf = sdk.ReadData("board0:/MMCComponents/ImageReadout_0", buf)
    if res == 0:
        for i in range(0, int(buf.info.valid_data)):
            print(f'{buf.data[i].pixel[0]:<8}', f'{buf.data[i].pixel[1]:<8}', f'{buf.data[i].pixel[2]:<8}')
            print(f'{buf.data[i].pixel[3]:<8}', f'{buf.data[i].pixel[4]:<8}', f'{buf.data[i].pixel[5]:<8}')    
            print(f'{buf.data[i].pixel[6]:<8}', f'{buf.data[i].pixel[7]:<8}', f'{buf.data[i].pixel[8]:<8}')
            print("")
    

File: examples/components/Java/Frame/src/main/java/io/github/nuclearinstruments/frame/Frame.java
package io.github.nuclearinstruments.frame;

import io.github.NuclearInstruments.jscisdk.*;

public class Frame {

    public static void main(String[] args) {
        SciSDK sdk = new SciSDK();

        // DT1260
        int res = sdk.AddNewDevice("usb:28645", "dt1260", "./DT1260RegisterFile.json", "board0");
        if (res != 0) {
            System.out.println("Program exit due to connection error");

            // print error description
            Ref<String> error_description = new Ref<>("");
            sdk.s_error(res, error_description);
            System.out.println("ERROR DESCRIPTION: " + error_description.get());
            return;
        }

        res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.thread", "false");
        res = sdk.SetParameterInteger("board0:/MMCComponents/ImageReadout_0.timeout", 500);
        res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.acq_mode", "non-blocking");
        res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.trigger_mode", "trig");
        res = sdk.SetParameterString("board0:/MMCComponents/ImageReadout_0.data_processing", "decode");
        
        // allocate buffer
        Ref<FrameDecodedBuffer> fdb = new Ref(new FrameDecodedBuffer());
        res = sdk.AllocateBuffer("board0:/MMCComponents/ImageReadout_0", fdb, 100);

        res = sdk.ExecuteCommand("board0:/MMCComponents/ImageReadout_0.start", "");

        // read data
        while (true) {
            res = sdk.ReadData("board0:/MMCComponents/ImageReadout_0", fdb);
            if (res == 0) {
                FramePacket[] fp = fdb.get().GetData();
                for (int i = 0; i < fp.length; i++) {
                    System.out.println(fp[i].GetPixel()[0] + ", " + fp[i].GetPixel()[1] + ", " + fp[i].GetPixel()[2]);
                    System.out.println(fp[i].GetPixel()[3] + ", " + fp[i].GetPixel()[4] + ", " + fp[i].GetPixel()[5]);
                    System.out.println(fp[i].GetPixel()[6] + ", " + fp[i].GetPixel()[7] + ", " + fp[i].GetPixel()[8]);
                    System.out.println("");
                }
            }
        }
    }
}
